networkit/cpp/matching/BSuitorMatcher.cpp

```diff
@@ -0,0 +1,152 @@
+#include <networkit/matching/BSuitorMatcher.hpp>
+
+#include <algorithm>
+#include <iomanip>
+#include <stdexcept>
+
+namespace NetworKit {
+BSuitorMatcher::BSuitorMatcher(const Graph &G, const std::vector<count> &b) : BMatcher(G, b), b(b) {
+    if (G.numberOfSelfLoops() > 0)
+        throw std::runtime_error("This algorithm does not support graphs with self-loops.");
+
+    if (G.isDirected())
+        throw std::runtime_error("This algorithm does not support directed graphs.");
+
+    if (b.size() != G.numberOfNodes())
+        throw std::runtime_error(
+            "The number of b values does not match the number of nodes in this graph.");
+
+    const auto n = G.upperNodeIdBound();
+    suitors.reserve(n);
+    proposed.reserve(n);
+    for (index i = 0; i < n; ++i) {
+        suitors.emplace_back(std::make_unique<MatchingNodeInfo>(b.at(i)));
+        proposed.emplace_back(std::make_unique<MatchingNodeInfo>(b.at(i)));
```

Comment:
I think it will break. Implemented now a simple check. 

In general many algorithms do not behave well, if the graph has node id holes. We should think about a general structure to fix this. I can see different strategies here: provide an internal mechanism with node id maps or having a general check mechanism in order to remove the need for implementing sanity checks for each algorithm.